cpp/benchmarks/io/text/multibyte_split.cpp

```diff
@@ -154,11 +156,17 @@ static void bench_multibyte_split(nvbench::state& state,
 
   auto source = [&] {
     switch (source_type) {
-      case data_chunk_source_type::file: {
+      case data_chunk_source_type::file:
+      case data_chunk_source_type::file_datasource: {
         auto const temp_file_name = random_file_in_dir(temp_dir.path());
         std::ofstream(temp_file_name, std::ofstream::out)
           .write(host_input.data(), host_input.size());
-        return cudf::io::text::make_source_from_file(temp_file_name);
+        if (source_type == data_chunk_source_type::file) {
+          return cudf::io::text::make_source_from_file(temp_file_name);
+        } else {
+          datasource = cudf::io::datasource::create(temp_file_name);
+          return cudf::io::text::make_source(*datasource);
```

Comment:
This seems dangerous. We get a `std::unique_ptr<datasource>` from `create` and then dereference that and pass it in as a `datasource` reference to `make_source`, which creates a `datasource_chunk_source` which holds a pointer to the `datasource`. The original `datasource` is going out of scope here. Am I missing something that keeps this data alive?

Reply:
`datasource` lives on the function scope, so it should survive. Note that this is not `auto datasource`

Reply:
Might be safer to pass a shared pointer to `make_source`. I'm thinking of the same change for the user-implemented datasource API.

Reply:
Sorry, I missed that this was a lambda. I saw the return and got worried.